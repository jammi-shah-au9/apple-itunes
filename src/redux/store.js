import { createStore,applyMiddleware } from 'redux'
import reducer from './reducers/root_reducer'
import { composeWithDevTools } from 'redux-devtools-extension'
import createSagaMiddleware from 'redux-saga'
import { watch_fetch_artist } from './sagas/saga'

const sagaMiddleware=createSagaMiddleware()
const store = createStore(reducer,composeWithDevTools(applyMiddleware(sagaMiddleware)))
sagaMiddleware.run(watch_fetch_artist)

export default store