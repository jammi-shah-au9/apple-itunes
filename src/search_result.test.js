import React from 'react';
import ReactDOM from 'react-dom'
import { Provider } from 'react-redux'
import store from './redux/store' 
import { render,cleanup } from '@testing-library/react';
import Searchresult from './search_result';
import { MemoryRouter as Router } from 'react-router-dom'
import renderer from 'react-test-renderer'

afterEach(cleanup)

it("renders without crashing",()=>{
  const div = document.createElement("div")
  ReactDOM.render(<Router><Provider store={store}><Searchresult /></Provider></Router>,div)
})


it("matches app snapshot",()=>{
  const apptree = renderer.create(<Router><Provider store={store}><Searchresult /></Provider></Router>).toJSON();
  expect(apptree).toMatchSnapshot();
})


