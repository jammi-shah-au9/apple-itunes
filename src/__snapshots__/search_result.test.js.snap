// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches app snapshot 1`] = `
Array [
  <div
    className="sc-gtssRu gfLRqM"
    data-testid="nav"
  >
    <div
      className="sc-bdnylx bQSunl"
    >
      <nav>
        <div>
          <i
            aria-hidden="true"
            className="fa fa-apple"
            id="logo"
          />
        </div>
        <div>
          Mac
        </div>
        <div>
          iPad
        </div>
        <div>
          iPhone
        </div>
        <div>
          Watch
        </div>
        <div>
          TV
        </div>
        <div>
          Music
        </div>
        <div>
          Support
        </div>
        <div>
          Where to Buy
        </div>
        <div>
          <i
            aria-hidden="true"
            className="fa fa-search"
          />
        </div>
      </nav>
    </div>
  </div>,
  <center>
    <h1
      className="sc-dlnjPT gREnMS"
    >
      
    </h1>
    <a
      className="sc-hKFyIo jJjecs"
      onClick={[Function]}
    >
      click to search again...
    </a>
  </center>,
  <center>
    <div
      className="sc-eCApGN fkszar"
    >
      <div
        className="sc-jSFkmK gPTwUW"
      />
    </div>
  </center>,
]
`;
